packaging/linux/fedora/sunshine.spec

```diff
@@ -0,0 +1,252 @@
+
+    # we need to patch the math-vector.h file for aarch64 fedora
+    # back up /usr/include/bits/math-vector.h
+    math_vector_file="/usr/include/bits/math-vector.h"
+
+    # patch headers https://bugs.launchpad.net/ubuntu/+source/mumax3/+bug/2032624
+    cp "$math_vector_file" "$math_vector_file.bak"
+    sed -i 's/__Float32x4_t/int/g' "$math_vector_file"
+    sed -i 's/__Float64x2_t/int/g' "$math_vector_file"
+    sed -i 's/__SVFloat32_t/float/g' "$math_vector_file"
+    sed -i 's/__SVFloat64_t/float/g' "$math_vector_file"
+    sed -i 's/__SVBool_t/int/g' "$math_vector_file"
+  fi
+
+  local url="${cuda_prefix}${cuda_version}/local_installers/cuda_${cuda_version}_${cuda_build}_linux${cuda_suffix}.run"
+  echo "cuda url: ${url}"
+  wget "$url" --progress=bar:force:noscroll -q --show-progress -O "%{_builddir}/cuda.run"
+  chmod a+x "%{_builddir}/cuda.run"
+  "%{_builddir}/cuda.run" --silent --toolkit --toolkitpath="%{_builddir}/cuda" --no-opengl-libs --no-man-page --no-drm
+  rm "%{_builddir}/cuda.run"
+}
+
+# we need to clear these flags to avoid linkage errors with cuda-gcc-c++
+export CFLAGS=""
+export CXXFLAGS=""
+export FFLAGS=""
+export FCFLAGS=""
+export LDFLAGS=""
+
+if [ -n "$cuda_version" ] && [[ " ${cuda_supported_architectures[@]} " =~ " ${architecture} " ]]; then
+  install_cuda
+  cmake_args+=("-DSUNSHINE_ENABLE_CUDA=ON")
+  cmake_args+=("-DCMAKE_CUDA_COMPILER:PATH=%{_builddir}/cuda/bin/nvcc")
+fi
+
+# cmake
+cd %{_builddir}/Sunshine
+echo "cmake args:"
+echo "${cmake_args[@]}"
+cmake "${cmake_args[@]}"
+make -j$(nproc) -C "%{_builddir}/Sunshine/build"
+
+%check
+# setup Xvfb
+export DISPLAY=:99
+Xvfb ${DISPLAY} -screen 0 1024x768x24 &
+
+# run tests
+cd %{_builddir}/Sunshine/build
+./tests/test_sunshine
+
+# teardown Xvfb
+Xvfb -kill ${DISPLAY}
+
+%install
+cd %{_builddir}/Sunshine/build
+%make_install
+
+# Add modules-load configuration
+install -D -m 0644 /dev/stdin %{buildroot}/usr/lib/modules-load.d/uhid.conf <<EOF
+uhid
+EOF
+
+%post
+# TODO: can we just use postinst script directly?
+# Ensure Sunshine can grab images from KMS
+path_to_setcap=$(which setcap)
+path_to_sunshine=$(readlink -f $(which sunshine))
+if [ -x "$path_to_setcap" ] ; then
+  echo "$path_to_setcap cap_sys_admin+p $path_to_sunshine"
+        $path_to_setcap cap_sys_admin+p $path_to_sunshine
+fi
+
+# Trigger udev rule reload for /dev/uinput and /dev/uhid
+path_to_udevadm=$(which udevadm)
+if [ -x "$path_to_udevadm" ] ; then
+  $path_to_udevadm control --reload-rules
+  $path_to_udevadm trigger --property-match=DEVNAME=/dev/uinput
+  $path_to_udevadm trigger --property-match=DEVNAME=/dev/uhid
+fi
+
+%preun
+# Remove udev rule
+rm -f %{_udevrulesdir}/99-uhid.rules  # TODO: why is this removed... doesn't look like a Sunshine file
+
+# Remove modules-load configuration
+rm -f /usr/lib/modules-load.d/uhid.conf  # TODO: why is this removed?
```

Comment:
this gets created during the .spec itself and i had issues with it lingering afterwards after removing sunshine. 

```
# Add modules-load configuration
install -D -m 0644 /dev/stdin %{buildroot}/usr/lib/modules-load.d/uhid.conf <<EOF
uhid
EOF
```
this is done to load the uhid module in initramfs even if it doesn't detect the module as being used during dracut, which must be run every time a new kernel is installed. from what i remember when packaging it, a user might otherwise need to `modprobe uhid` manually.